TaskLayer/SearchTask/PostSearchAnalysisTask.cs

```diff
@@ -859,5 +870,102 @@ private void WritePeakQuantificationResultsToTsv(FlashLfqResults flashLFQResults
 
             FinishedWritingFile(peaksPath, nestedIds);
         }
+
+        private void DetermineBestCategoryPsms()
+        {
```

Comment:
I don't really get this. In the FDR engine, can you just do `AllPsms.GroupBy(p => p.BestPeptide.Peptide.CleavageSpecificity)` in the FDR analysis engine, and then for each group, calculate group q-values? Seems much simpler, what am I missing?

How do you want to handle ambiguity? Like if a spectrum matches to a semi-specific peptide and a full specificity peptide, what category does it belong in for FDR purposes?

Reply:
We're not getting the best psms from each category, but rather a single psm for each scan and discriminating based on the category specific q-value.
Because of this, when a spectrum matches to a semi- and a fully-specific peptide, the one with the lower q-value wins out.